shelley/chain-and-ledger/formal-spec/epoch.tex

```diff
@@ -900,6 +900,8 @@ \subsection{Complete Epoch Boundary Transition}
       \\~\\~\\
       \var{(\wcard,~\wcard,~\wcard,~(\var{pup},~\wcard,~\wcard,~\wcard))}\leteq\var{utxoSt'}\\
       \var{pp_{new}}\leteq\fun{votedValue_{PParams}}~\var{pup}\\
+      % TODO: I think this isn't the right type, `pup` is PPUpdate which is not
+      % a complete set of `PParams`.
```

Comment:
I think it is okay, though it maybe does not translate well to Haskell. The type of `PParams` is a mapping of `PPm` to `Value`.

Reply:
ok, I see. So that's a place where the executable spec deviates a bit `PParams` is a record with all values and the parameter update is a mapping. This comes in handy whenever a value of `PParams` is read as the mapping would require checking for presence of the element. I will create an issue for that.